src/components/common/Banner/utils.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    ButtonComponentType,
+    ButtonProps,
+    ButtonStyleType,
+    ButtonVariantType,
+    ComponentSizeType,
+    getDateInMilliseconds,
+    Icon,
+    IconsProps,
+    InfoBlockProps,
+    InfoBlockVariant,
+    InfoBlockVariantType,
+    refresh,
+    VARIANT_TO_BG_MAP,
+    VARIANT_TO_ICON_COLOR_MAP,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { ANNOUNCEMENT_CONFIG, BannerVariant } from './constants'
+import { BannerConfigProps, BannerConfigType } from './types'
+
+export const getBannerIcon = (
+    bannerVariant: BannerVariant,
+    type: InfoBlockProps['variant'],
+    iconName: IconsProps['name'],
+) => {
+    const variantWithIconMap: Record<BannerVariant, IconsProps['name'] | null> = {
+        [BannerVariant.INTERNET_CONNECTIVITY]: 'ic-disconnect',
+        [BannerVariant.VERSION_UPDATE]: 'ic-sparkle-color',
+        [BannerVariant.LICENSE]: iconName,
+        [BannerVariant.ANNOUNCEMENT]: 'ic-megaphone-left',
+    }
+
+    return (
+        <Icon
+            name={variantWithIconMap[bannerVariant]}
+            color={bannerVariant === BannerVariant.ANNOUNCEMENT ? VARIANT_TO_ICON_COLOR_MAP[type] : null}
+            size={16}
+        />
+    )
+}
+
+export const getBannerConfig = ({
+    bannerVariant,
+    isOnline,
+    licenseType,
+    enterpriseLicenseBarMessage = '',
+    hideInternetConnectivityBar = false,
+}: BannerConfigProps): BannerConfigType => {
+    const bannerConfigMap: Partial<Record<BannerVariant, BannerConfigType>> = {
```

Comment:
we should change this to switch case instead creating a new map or may be we can create map outside this method and based on condition return from that or internetConnectivity